src/etc/inc/interfaces.inc

```diff
@@ -3195,88 +3309,202 @@ function interface_dhcpv6_configure($interface = 'wan', $wancfg)
+        $dhcp6cscript .= "\t;;\n";
+        $dhcp6cscript .= "*)\n";
+        $dhcp6cscript .= "\t;;\n";
+        $dhcp6cscript .= "esac\n";
+    
+        if (!@file_put_contents("/var/etc/dhcp6c_{$interface}_script.sh", $dhcp6cscript)) {
+            printf("Error: cannot open dhcp6c_{$interface}_script.sh in interface_dhcpv6_configure() for writing.\n");
+            unset($dhcp6cscript);
+            return 1;
+        } else { 
+            chmod("/var/etc/dhcp6c_{$interface}_script.sh", 0755);
+        }
     }
+    unset($dhcp6cscript);
 
-    return 0;
+}
+
+function DHCP6_Config_File_Basic($interface,$wancfg)
+{
+    global $config;
+    
+    $wanif = get_real_interface($interface, "inet6");
+    $iflist = link_interface_to_track6($interface);
+    $dhcp6c_conf = "";
+    
+     if ($wancfg['ipaddrv6'] == "slaac")  {
+        $dhcp6c_conf .= "interface {$wanif} {\n";
+        /* for SLAAC interfaces we do fire off a dhcp6 client for just our name servers */
+        $dhcp6c_conf .= "  information-only;\n";
+        $dhcp6c_conf .= "  request domain-name-servers;\n";
+        $dhcp6c_conf .= "  request domain-name;\n";
+        $dhcp6c_conf .= "  script \"/var/etc/dhcp6c_{$interface}_script.sh\"; # we'd like some nameservers please\n";
+        $dhcp6c_conf .= "};\n";
+    
+    } else {
+        $dhcp6c_conf .= "interface {$wanif} {\n";
+        $dhcp6c_conf .= "  send ia-na 0; # request stateful address for $friendly\n";
+        $dhcp6c_conf .= "  send ia-pd 0; # request prefix delegation for $friendly\n";
+        $dhcp6c_conf .= "  request domain-name-servers;\n";
+        $dhcp6c_conf .= "  request domain-name;\n";
+        $dhcp6c_conf .= "  script \"/var/etc/dhcp6c_{$interface}_script.sh\"; # we'd like some nameservers please\n";
+        $dhcp6c_conf .= "};\n";
+  
+        $count = 0;
+        
+        if (!isset($wancfg['dhcp6prefixonly'])) {
+                $dhcp6c_conf .= "id-assoc na 0 { };\n";
+        }
+        $dhcp6c_conf .= "id-assoc pd 0 {\n"; 
+        $preflen = 64 - $wancfg['dhcp6-ia-pd-len'];
+        if (isset($wancfg['dhcp6-ia-pd-send-hint'])) {
+            $dhcp6c_conf .= "  prefix ::/{$preflen} infinity;\n";
+        }
+        foreach ($iflist as $friendly => $ifcfg) {
+            if (is_numeric($wancfg['dhcp6-ia-pd-len'])) {
+                /* Setup the prefix delegation */ 
+                if (isset($wancfg['dhcp6incslaid'])) {
+                    $sla_id = $count;
+                } else {
+                    $sla_id = 0;
+                }
+                if (is_numeric($ifcfg['track6-prefix-id'])) {
+                    $realif = get_real_interface($friendly);
+                    $dhcp6c_conf .= "  prefix-interface {$realif} {\n";
+                    $dhcp6c_conf .= "    sla-len {$wancfg['dhcp6-ia-pd-len']};\n";
+                    $dhcp6c_conf .= "    sla-id {$sla_id};\n";
```

Comment:
is there any reason why we omit track6-prefix-id usage here now? that's a detail I don't want to change without further tweaking, e.g. allowing the value to be empty in the GUI so this can be auto-assigned?

Reply:
No, that's a mistake on my part, it should follow the config. Sorry, Are you OK to correct that?

Reply:
I will cherry-pick and modify a tiny bit then, no need to change this PR. Ok? :)

Reply:
Your're comment elsewhere about dhcp6c being slow to exit reminded me of the reason for the  addition of the USR1 sig,

Reply:
That's cool Franco. :)

Reply:
👍

Reply:
I wanted to use a php call to the run_dhcp6client_process() from within the rtsold script, but although I could pass the params, I had to include util.inc and even then it would not correctly get all the variables. It would have been nice though to have only one place where dhcp6c can be launched.